lib/ansible/module_utils/urls.py

```diff
@@ -734,7 +734,12 @@ def http_request(self, req):
             if https_proxy:
                 proxy_parts = generic_urlparse(urlparse(https_proxy))
                 port = proxy_parts.get('port') or 443
-                s = socket.create_connection((proxy_parts.get('hostname'), port))
+                proxy_hostname = proxy_parts.get('hostname', None)
+                if proxy_hostname is None or proxy_parts.get('scheme') == '':
```

Comment:
Would it not make sense to assume 'https' when port is 443 ?

Reply:
Yes. But still `generic_urlparse` return proxy_hostname as None so it does not make sense to proceed further.

Reply:
Ah, ok :-)